### ISSUE TITLE
Postgres enhancements to reduce replication issues and speed up restoration

### ISSUE THREAD

**Issue by RobHooper** [open-contracting/deploy#176]
These are improvements I noted while investigating and restoring broken replication between the kingfisher servers this morning.

* We are currently storing 10 wal logs, this is fine most of the time, however last night there was a huge amount of data written to kingfisher-process1 and it failed to keep up.
* Setting up postgres "replica slots" that should force the master server to hold onto wal logs until the replica gives it the OK.
* We are archiving old wal logs - which I was able to use to restore kingfisher-replica1 from. This directory is very big and we should start clearing it down to speed up restores. - I propose keeping the last week worth of wal logs.
* We could also start proactively copying archived wal logs between servers, then postgres will be able to automatically restore from them should this happen again.

**Comment by RobHooper** [comment]
Increasing the `max_wal_size` will also help with item 1. This already has an issue #158

**Comment by jpmckinney** [comment]
Sounds good to me. Would just replica slots (point 2) be enough?

**Comment by RobHooper** [comment]
Replica slots may well be enough to keep replication working, however I'm not sure yet how/if we can automate setting this up.
The others are more nice to haves, changing the stored wal logs + wal log size is a quick win as they are just config settings.
Clear down the wal archive is the next priority as it could lead to the disk slowly filling up.

**Comment by RobHooper** [comment]
Raised PR #182 which addresses point 3 / archiving old wal logs

Regarding replica slots. I have looked into moving them into salt and decided to leave this as a manual step for now.
Most of the changes for replica slots are custom to the two servers we've set it up on and I'm not sure of the best way to make this a "template" and reusable for future servers at the moment. I suggest picking this up again when new postgres replicas come online.

I've documented this set up at my end. 
For reference, these are the postgres replica slot specific commands I ran:
```
# On kingfisher-replica1
echo "primary_slot_name = 'replica1'" >> /var/lib/postgresql/11/main/recovery.conf

# On kingfisher-process1
psql
select * from pg_create_physical_replication_slot('replica1');
```
The rest of recovery.conf is autogenerated by pg_basebackup - [see docs](https://ocdsdeploy.readthedocs.io/en/latest/deploy/postgres.html)


Replica slots have been working well so I see no need for any additional fixes (proactively copying wal logs to the replica).

**Comment by jpmckinney** [comment]
@RobHooper Sounds good. Can you document the location of `recovery.conf` to allow for manual inspection?

> I've documented this set up at my end.

Was there anything else in addition to your earlier comment?

If we can add these notes to postgres.rst, I'm happy to then close this issue once #182 is closed.

**Comment by RobHooper** [comment]
Documentation was updated.

I've merged and deployed PR #182 addressing these remaining issues.
I also manually ran the clean up script to resolve and disk usage issues sooner.

**Comment by jpmckinney** [comment]
Woohoo, down to 40% from 70% drive usage :) 🎉
